packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmImpl.kt

```diff
@@ -225,7 +226,8 @@ public class RealmImpl private constructor(
     }
 
     override fun <T : CoreNotifiable<T, C>, C> registerObserver(t: Observable<T, C>, keyPaths: Pair<ClassKey, List<String>>?): Flow<C> {
-        return notifier.registerObserver(t, keyPaths)
+        val keypathsPtr: RealmKeyPathArrayPointer? = keyPaths?.let { RealmInterop.realm_create_key_paths_array(realmReference.dbPointer, keyPaths.first, keyPaths.second) }
```

Comment:
Yes, it is the same owner, the `realm` instance in the Notifier is thread-confined, and in this case we are calling it outside the notifier dispather. You can see that the rest of the code in registerObserver is run on the correct dispatcher.

My guess is that it was a race condition since the Notifier Realm is lazy-created, so depending on which thread created the Realm, we could accidentally get a thread-exception here.

Reply:
Ahh, yeah.